src/add_toppra_time_parameterization.cpp

```diff
@@ -131,100 +152,91 @@ class AddToppraTimeParameterization : public planning_interface::PlanningRespons
-      }
-      else
-      {
-        RCLCPP_ERROR_STREAM(getLogger(), "No acceleration limit was defined for joint "
-                                             << joint_variables.at(idx).c_str()
-                                             << "! You have to define "
-                                                "acceleration "
-                                                "limits in the URDF or "
-                                                "joint_limits.yaml");
-        res.error_code = moveit::core::MoveItErrorCode::FAILURE;
-        return;
-      }
-    }
-    toppra.AddJointVelocityLimit(max_velocity, min_velocity);
-    toppra.AddJointAccelerationLimit(min_acceleration, max_acceleration);
-    auto optimized_trajectory = toppra.SolvePathParameterization();
+    // Run toppra
+    const auto grid_points = Toppra::CalcGridPoints(input_path, CalcGridPointsOptions());
+    auto toppra = Toppra(input_path, plant, grid_points);
+
+    // Get velocity and acceleration bounds
+    Eigen::VectorXd lower_velocity_limits;
+    Eigen::VectorXd upper_velocity_limits;
+    Eigen::VectorXd lower_acceleration_limits;
+    Eigen::VectorXd upper_acceleration_limits;
+
+    getVelocityBounds(joint_model_group, plant, lower_velocity_limits, upper_velocity_limits);
+    getAccelerationBounds(joint_model_group, plant, lower_acceleration_limits, upper_acceleration_limits);
 
-    if (!optimized_trajectory.has_value())
+    toppra.AddJointVelocityLimit(lower_velocity_limits, upper_velocity_limits);
+    toppra.AddJointAccelerationLimit(lower_acceleration_limits, upper_acceleration_limits);
+    auto time_optimized_path_parameterization = toppra.SolvePathParameterization();
+
+    if (!time_optimized_path_parameterization.has_value())
     {
       RCLCPP_ERROR_STREAM(getLogger(), "Failed to calculate a trajectory with toppra");
       res.error_code = moveit::core::MoveItErrorCode::FAILURE;
       return;
     }
 
-    getRobotTrajectory(optimized_trajectory.value(), res.trajectory->getWayPointCount() /* TODO enable down sampling */,
-                       res.trajectory /* override previous solution with optimal trajectory*/);
+    // create optimized trajectory
+    auto optimized_trajectory =
+        PathParameterizedTrajectory<double>(input_path, time_optimized_path_parameterization.value());
+
+    // Transfer optimized trajectory back to moveit trajectory
+    getRobotTrajectory(optimized_trajectory,
+                       optimized_trajectory.end_time() /
+                           res.trajectory->getWayPointCount() /* TODO enable down sampling */,
+                       plant, res.trajectory /* override previous solution with optimal trajectory*/);
+
+    // meshcat experiment
+    auto& vis_context = visualizer_->GetMyContextFromRoot(*diagram_context_);
+    visualizer_->ForcedPublish(vis_context);
+
+    // Visualize the trajectory with Meshcat (uncomment to visualize)
```

Comment:
Ideally this would be a parameter or something, but not necessary now

Reply:
Yes, I need to add a parameter API like in KTOpt in a follow-up PR